pkg/sql/catalog/internal/catkv/catalog_reader.go

```diff
@@ -24,20 +24,97 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 )
 
-// catalogReader builds catalogQuery objects and runs them, leveraging the
-// SystemDatabaseCache if it is present.
+// CatalogReader queries the system.namespace and system.descriptor tables,
+// leveraging the SystemDatabaseCache if it is present in the implementation.
+//
+// The main use case for CatalogReader should be StoredCatalog.
+type CatalogReader interface {
```

Comment:
I gave this a proper interface to encapsulate all KV operations. It's not strictly-speaking necessary for this commit but it's a nice change so I left it in.